pkg/util/sync/wait_with_timeout.go

```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package sync
+
+import (
+	"fmt"
+	"sync"
+	"time"
+)
+
+func WaitWithTimeout(wg *sync.WaitGroup, timeout time.Duration) error {
+	done := make(chan struct{})
+	go func() {
+		wg.Wait()
+		close(done)
+	}()
+
+	select {
+	case <-time.After(timeout):
+		return fmt.Errorf("WaitGroup.Wait() timed out after %s", timeout)
```

Comment:
there's a race between waiting the wait group and returning an error. not sure if this won't cause problems. maybe document that WaitWithTimeout may return an error even if the WaitGroup.Wait() returns before the timeout expires

Reply:
Ugh... I don't think that's worth documenting, just the call to `WaitWithTimeout` already makes it being imprecise. The idea of waiting on waitgroup with a timeout is to avoid blocking forever on something we expect to take a limited amount of time, not to be precise on how long it took (waitgroups would be a bad choice for that)